File: 2059-unique-length-3-palindromic-subsequences/unique-length-3-palindromic-subsequences.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        count = 0

        for i in range(26):
            left, right = s.find(chr(i + 97)), s.rfind(chr(i + 97))
            if left != -1 and right != -1:
                count += len(set(s[left + 1 : right]))

        return count

File: 2059-unique-length-3-palindromic-subsequences/unique-length-3-palindromic-subsequences.cs
public class Solution {
    public int CountPalindromicSubsequence(string s) {
        string alpha = "abcdefghijklmnopqrstuvwxyz";
        int count = 0, left, right;
        HashSet<char> hash = new HashSet<char>();

        foreach (char c in alpha) {
            left = s.IndexOf(c);
            right = s.LastIndexOf(c);

            if (left != -1 && right != -1 && left < right) {
                for (int i = left + 1; i < right; i++) {
                    if (!hash.Contains(s[i])) {
                        count++;
                        hash.Add(s[i]);
                    }
                    if (hash.Count == 26) {
                        break;
                    }
                }
            }

            hash.Clear();
        }

        return count;
    }
}
